credsweeper/file_handler/text_content_provider.py

```diff
@@ -23,5 +26,13 @@ def get_analysis_target(self) -> List[AnalysisTarget]:
             list of analysis targets based on every row in file
 
         """
-        lines = Util.read_file(self.file_path)
+        lines = None
+
+        if Util.get_extension(self.file_path) == ".xml":
+            lines, line_nums = Util.get_xml_data(self.file_path)
+            if lines:
+                return self.lines_line_num_to_targets(lines, line_nums)
+
+        if lines is None:
```

Comment:
'if' may be extra due there was return before

Reply:
Actually it is needed.
But I refactored the code for more readability.
Please check f02995f.